plugins/kotlin/code-insight/kotlin.code-insight.k2/src/org/jetbrains/kotlin/idea/k2/codeinsight/quickFixes/createFromUsage/K2ElementActionsFactory.kt

```diff
@@ -114,9 +118,189 @@ class K2ElementActionsFactory : JvmElementActionsFactory() {
         }
         return actions
     }
+
+    override fun createChangeAnnotationAttributeActions(annotation: JvmAnnotation,
+                                                        attributeIndex: Int,
+                                                        request: AnnotationAttributeRequest,
+                                                        @IntentionName text: String,
+                                                        @IntentionFamilyName familyName: String): List<IntentionAction> {
+        val annotationEntry = ((annotation as? KtLightElement<*, *>)?.kotlinOrigin as? KtAnnotationEntry)?.takeIf {
+            it.language == KotlinLanguage.INSTANCE
+        } ?: return emptyList()
+
+        return listOf(ChangeAnnotationAction(annotationEntry, attributeIndex, request, text, familyName))
+    }
+
+    //TODO: this is copy paste from KotlinElementActionsFactory, maybe we should extract it to common place?
+    // or implement this for K1 and K2?
+
+    private class ChangeAnnotationAction(
+        annotationEntry: KtAnnotationEntry,
+        private val attributeIndex: Int,
+        private val request: AnnotationAttributeRequest,
+        @IntentionName private val text: String,
+        @IntentionFamilyName private val familyName: String
+    ) : IntentionAction {
```

Comment:
in K2 mode, we try to implement intentions through `org.jetbrains.kotlin.idea.codeinsight.api.applicable.intentions.KotlinPsiUpdateModCommandAction`. You won't need to implement preview and it's easier to reuse the code for different headless applications

Reply:
Okay but I can't extract this class to `kotlin.code-insight.intentions.k2`